torch/_inductor/ir.py

```diff
@@ -6390,21 +6390,25 @@ def mark_reuse(self, users):
         that is used multiple times.
         """
 
-        def should_realize_on_cpu(loops: Union[Pointwise, Reduction]):
+        def should_realize_on_cpu(loops: Union[Pointwise, Reduction], users):
             """
             The heuristic for realizing reused result of heavy ops on cpu
             """
             heavy_ops = ["exp", "sigmoid"]  # a list of heavy ops
             fn_str = loops.inner_fn_str()
-            return any((op + "(") in fn_str for op in heavy_ops)
+            multi_users = (
+                users > 5
+                and self.inner_fn_opcount() > config.realize_opcount_threshold - 2
```

Comment:
Why -2?

Reply:
This is a heuristic update. We want to set checks to meet the requirements of doing realize when users is 6 and self.inner_fn_opcount() is 29.